pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -107,22 +109,32 @@ func (m *kubeGenericRuntimeManager) generateLinuxContainerConfig(container *v1.C
 			unified[cm.MemoryMin] = strconv.FormatInt(memoryRequest, 10)
 		}
 
-		// If container sets limits.memory, we set memory.high=pod.spec.containers[i].resources.limits[memory] * memory_throttling_factor
-		// for container level cgroup if memory.high>memory.min.
-		// If container doesn't set limits.memory, we set memory.high=node_allocatable_memory * memory_throttling_factor
-		// for container level cgroup.
-		memoryHigh := int64(0)
-		if memoryLimit != 0 {
-			memoryHigh = int64(float64(memoryLimit) * m.memoryThrottlingFactor)
-		} else {
-			allocatable := m.getNodeAllocatable()
-			allocatableMemory, ok := allocatable[v1.ResourceMemory]
-			if ok && allocatableMemory.Value() > 0 {
-				memoryHigh = int64(float64(allocatableMemory.Value()) * m.memoryThrottlingFactor)
+		// Guaranteed pods by their QoS definition require memory requests=memory limits and are not overcommitted.
+		// Hence MemoryQoS feature is disabled on those pods by not setting memory.high.
+		if memoryRequest != memoryLimit {
+			// The formula for memory.high for container cgroup is modified in Alpha stage of the feature in K8s v1.27.
+			// It will be set based on formula:
+			// `memory.high=floor[(requests.memory + memory throttling factor * (limits.memory or node allocatable memory - requests.memory))/pageSize] * pageSize`
+			// where default value of memory throttling factor is set to 0.9
+			// More info: https://git.k8s.io/enhancements/keps/sig-node/2570-memory-qos
+			memoryHigh := int64(0)
+			pageSize := int64(os.Getpagesize())
```

Comment:
nit, can we consider to cache the page size maybe just making a global variable under the package or using a `sync.Once`, i.e.

```
var defaultPageSize = os.Getpagesize()
```

otherwise, we have to make the getpagesize syscall each time here, which is unnecessary since the page size will not change during runtime

Reply:
Good Point.